kythe/rust/indexer/src/indexer/analyzers.rs

```diff
@@ -751,42 +825,60 @@ impl<'a, 'b> CrateAnalyzer<'a, 'b> {
             }
             reference_vname.set_path(file_vname.unwrap().get_path().to_string());
 
-            // Get byte span
-            let start_byte_option = self.offset_index.get_byte_offset(
-                span.file_name.to_str().unwrap(),
-                span.line_start.0,
-                span.column_start.0,
-            );
-
-            // If the start byte is none, then the save_analysis is giving information about
-            // standard library files and we should skip
-            if start_byte_option.is_none() {
+            let byte_offsets_option = self.get_byte_offsets(&reference.ref_id, span)?;
+            if byte_offsets_option.is_none() {
                 continue;
             }
-
-            let start_byte = start_byte_option.unwrap();
-            let end_byte = self
-                .offset_index
-                .get_byte_offset(
-                    span.file_name.to_str().unwrap(),
-                    span.line_end.0,
-                    span.column_end.0,
-                )
-                .ok_or_else(|| {
-                    KytheError::IndexerError(format!(
-                        "Failed to get ending offset for reference {:?}",
-                        reference
-                    ))
-                })?;
+            let byte_offsets = byte_offsets_option.unwrap();
 
             // Create signature based on span
-            reference_vname
-                .set_signature(format!("{}_ref_{}_{}", krate_signature, start_byte, end_byte));
-
-            self.emitter.emit_reference(&reference_vname, &target_vname, start_byte, end_byte)?;
+            reference_vname.set_signature(format!(
+                "{}_ref_{}_{}",
+                krate_signature, byte_offsets.start_byte, byte_offsets.end_byte
+            ));
+
+            self.emitter.emit_reference(
+                &reference_vname,
+                &target_vname,
+                byte_offsets.start_byte,
+                byte_offsets.end_byte,
+            )?;
         }
         Ok(())
     }
+
+    /// Return byte offsets for a span
+    fn get_byte_offsets(
+        &self,
+        ref_id: &rls_data::Id,
+        span: &rls_data::SpanData,
+    ) -> Result<Option<ByteOffsets>, KytheError> {
+        // Get byte span
+        let start_byte_option = self.offset_index.get_byte_offset(
+            span.file_name.to_str().unwrap(),
+            span.line_start.0,
+            span.column_start.0,
+        );
+
+        // If the start byte is none, then the save_analysis is giving information about
+        // standard library files and we should skip
+        if start_byte_option.is_none() {
```

Comment:
Is there a verifier test for this case? Can you add one?

Reply:
https://kythe.io/docs/kythe-verifier.html section "Checking that rules are unsatisfiable"